src/System.Management/src/System/Management/ManagementPath.cs

```diff
@@ -322,9 +322,9 @@ private static string GetWbemPath(IWbemPath wbemPath)
 
                     if (status >= 0 && 0 < bufLen)
                     {
-                        StringBuilder pathBuilder = new StringBuilder((int)bufLen - 1);
-                        status = wbemPath.GetText_(flags, ref bufLen, pathBuilder);
-                        pathStr = pathBuilder.ToString();
+                        char[] pathChars = new char[(int)bufLen - 1];
+                        status = wbemPath.GetText_(flags, ref bufLen, pathChars);
+                        pathStr = new string(pathChars);
```

Comment:
This isn't quite right (same for the other ctor uses).  It'll need to instead be something like:
```C#
new string(pathChars, 0, Array.IndexOf(pathChars, '\0'));
```